qiita_db/portal.py

```diff
@@ -5,13 +5,66 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+import warnings
 
 from .sql_connection import SQLConnectionHandler
 from .util import convert_to_id
+from .exceptions import QiitaDBError, QiitaDBDuplicateError
 from qiita_core.qiita_settings import qiita_config
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
 
 # Only make functions available for main portal
 if qiita_config.portal == "QIITA":
+    def create_portal(portal, desc):
```

Comment:
The other methods in here don't appear to have common state, so there is no instance state to manipulate, which would make them `staticmethod`s. A `property` can still be considered a `staticmethod` if it doesn't modify or access instance state. These could be coerced into instance methods if `portal` was stored but it doesn't really buy you much except for the namespace.

These methods could probably be coerced into a subclass of a `QiitaObject`, which I support simply for consistency. 

I disagree that the argument extends to the rest of the objects in the system as there are notions of common state, (sane) operator overloading and the encapsulation is pleasant. Keep in mind though, it was not necessary to use objects, and without seeing alternative implementations, it's difficult to evaluate if that was the best decision (I am in no way suggesting we revisit this). 

I still don't understand why these functions should be defined IFF the configuration file says 'qiita'. It sounds like the "portal manager" is really just a means of access control. If that is the case, then this approach is flawed.